db/migrate/20151203222446_create_media.rb

```diff
@@ -0,0 +1,13 @@
+class CreateMedia < ActiveRecord::Migration
+  def change
+    create_table :media do |t|
+      t.string :type
+      t.string :name
+      t.string :creator
+      t.string :description
+      t.integer :upvotes
```

Comment:
You could set a default value for this to be 0. This would simplify a lot of your later logic, which deals with incrementing numbers and would remove the 'nil' factor you encounter for media records that have not been upvoted yet.